doc/application/application.rst

```diff
@@ -641,6 +641,73 @@ Zephyr binary into your application directory.
 You can also define the ``SOC_ROOT`` variable in the application
 :file:`CMakeLists.txt` file.
 
+.. _ext-projs:
+
+Modules (External projects)
+***************************
+
+Zephyr relies on the source code of several externally maintained projects in
+order to avoid reinventing the wheel and to reuse as much well-established,
+mature code as possible when it makes sense. In the context of Zephyr's build
+system those are called *modules*.
+There are several categories of external projects that Zephyr depends on,
+among which:
+
+- Debugger integration
+- Silicon Vendor Hardware Abstraction Layers (HALs)
+- Cryptography libraries
+- Filesystems
+- Inter-Process Communication (IPC)
+
+These modules (external projects) must be integrated with the Zephyr
+build system, which is based around Kconfig and CMake (see
+:ref:`application` for more information about Zephyr's build system).
+
```

Comment:
Clarify module vs project:

- zephyr can never be a module
- a manifest project can be a module
...

Reply:
> Clarify module vs project:

will wait for this then re-review